reana_server/rest/users.py

```diff
@@ -0,0 +1,278 @@
+        admin = Session.query(User).filter_by(id_=ADMIN_USER_ID).one_or_none()
+        if token != admin.api_key:
+            return jsonify({"message": "Action not permitted."}), 403
+        search_criteria = dict()
+        if user_id:
+            search_criteria['id_'] = user_id
+        if user_email:
+            search_criteria['email'] = user_email
+        user = Session.query(User).filter_by(**search_criteria).one_or_none()
+        if user:
+            return jsonify(id_=user.id_,
+                           email=user.email,
+                           token=user.api_key), 200
+        else:
+            return jsonify({"message": "User {} does not exist.".
+                           format(user_id)}, 404)
+    except Exception as e:
+        logging.error(traceback.format_exc())
+        return jsonify({"message": str(e)}), 500
+
+
+@blueprint.route('/users', methods=['POST'])
+def create_user():  # noqa
+    r"""Endpoint to create users.
+
+    ---
+    post:
+      summary: Creates a new user with the provided information.
+      description: >-
+        This resource creates a new user with the provided
+        information (email, id). Requires the admin api key.
+      operationId: create_user
+      produces:
+        - application/json
+      parameters:
+        - name: email
+          in: query
+          description: Required. The email of the user.
+          required: true
+          type: string
+        - name: token
+          in: query
+          description: Required. API key of the admin.
+          required: true
+          type: string
+      responses:
+        201:
+          description: >-
+            User created successfully. Returns the token and a message.
+          schema:
+            type: object
+            properties:
+              id_:
+                type: string
+              email:
+                type: string
+              token:
+                type: string
+          examples:
+            application/json:
+              {
+                "id_": "00000000-0000-0000-0000-000000000000",
+                "email": "user@reana.info",
+                "token": "Drmhze6EPcv0fN_81Bj-nA"
+              }
+        403:
+          description: >-
+            Request failed. The incoming payload seems malformed.
+        500:
+          description: >-
+            Request failed. Internal server error.
+          examples:
+            application/json:
+              {
+                "message": "Internal server error."
+              }
+    """
+    try:
+        user_email = request.args.get('email')
+        token = request.args.get('token')
+        admin = Session.query(User).filter_by(id_=ADMIN_USER_ID).one_or_none()
+        if token != admin.api_key:
+            return jsonify({"message": "Action not permitted."}), 403
+        user_parameters = dict(api_key=secrets.token_urlsafe())
+        user_parameters['email'] = user_email
+        user = User(**user_parameters)
```

Comment:
Aren't we missing the workspace creation here? Like in L70 in `reana_server/cli.py`